ext/gd/gd.c

```diff
@@ -652,7 +652,20 @@ PHP_FUNCTION(imagesetstyle)
 	stylearr = safe_emalloc(sizeof(int), num_styles, 0);
 
 	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(styles), item) {
-		stylearr[index++] = zval_get_long(item);
+		bool failed = false;
+		ZVAL_DEREF(item);
+		zend_long tmp = zval_try_get_long(item, &failed);
+		if (failed) {
+			efree(stylearr);
+			zend_argument_value_error(2, "value must be of type int, %s given", zend_zval_type_name(item));
```

Comment:
Should be a type error? And maybe clarify that its the elements of the style array?